tools/codegen/model.py

```diff
@@ -1496,3 +1503,40 @@ def parse_returns(return_decl: str) -> Tuple[Return, ...]:
     if return_decl[0] == '(' and return_decl[-1] == ')':
         return_decl = return_decl[1:-1]
     return tuple(Return.parse(arg) for arg in return_decl.split(', '))
+
+@dataclass(frozen=True)
+class PrecomputedElement:
+    name: str
+    ty: str
+
+    @staticmethod
+    def build(src: str) -> 'PrecomputedElement':
+        ty, name = src.split(" ")
+        return PrecomputedElement(name=name, ty=ty)
```

Comment:
In a bunch of the other parsing code we have self validation code that looks like this:

```
        r = Return(
            name=name,
            type=type,
            annotation=annotation,
        )
        assert str(r) == arg, f'{str(r)} != {arg}'
        return r
```

consider adding it here too!

Reply:
Very cool, but how is `__str__` defined on `PrecomputedElement`? Is it added by the `@dataclass` decorator?